src/IssueViz.Security/Hotspots/LocalHotspotsStore.cs

```diff
@@ -0,0 +1,208 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using SonarLint.VisualStudio.IssueVisualization.Security.IssuesStore;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.Hotspots
+{
+    public interface ILocalHotspotsStoreUpdater
+    {
+        void AddOrUpdate(string filePath, IEnumerable<IAnalysisIssueVisualization> hotspots);
+        void RemoveForFile(string filePath);
+    }
+
+    internal interface ILocalHotspotsStore : ILocalHotspotsStoreUpdater, IIssuesStore
+    {
+        List<LocalHotspot> GetAllLocalHotspots();
```

Comment:
I've actually been following the opposite rule: accept abstract types where possible, return concrete types where possible. 

The idea with list here is that it's a copy each time (the elements are immutable), so that the list is not modified by the source after it's returned

Reply:
That's not the approach we follow anywhere else.

Effectively what you're saying is "because we're returning a copy we don't care if you change the list", which isn't quite the same thing as return a type that conveys "here is a read-only list".

We can still return a copy each time if we want to, but it's an implementation detail.

Reply:
OK, I see what you mean